### PR TITLE
convert deployment target to OS

### PR DESCRIPTION
**Pull request by compnerd**

### CODE REVIEW

File: CoreFoundation/Base.subproj/CFRuntime.c [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ Boolean __CFInitialized = 0;
 // move the next 2 lines down into the #if below, and make it static, after Foundation gets off this symbol on other platforms. 
 CF_EXPORT _CFThreadRef _CFMainPThread;
 _CFThreadRef _CFMainPThread = kNilPthreadT;
-#if DEPLOYMENT_TARGET_WINDOWS || DEPLOYMENT_TARGET_LINUX
+#if DEPLOYMENT_TARGET_WINDOWS || TARGET_OS_LINUX
```
Comment (millenomi) [inline CoreFoundation/Base.subproj/CFRuntime.c]: `DEPLOYMENT_TARGET_WINDOWS` is `TARGET_OS_WIN32`

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by millenomi** [comment]
Just the one note above, but good otherwise.

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**